### PR TITLE
Ipsec Policy for Virtual Network Gateway Connection

### PR DESCRIPTION
**Pull request by henry416**
## Description
**THIS CHANGE MUST GO IN EARLY APRIL RELEASE**

IPsec Policy support for Virtual Network Gateway Connection
+ updated Network nuget package (2017-03-01 api version)
+ updated existing Virtual Network Gateway Connection model + cmdlets
+ added IPsec Policy model + cmdlet
+ added connection test for IPsec Policy

Reference PR: https://github.com/Azure/azure-sdk-for-net/pull/2943
---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by henry416** [comment]
This PR is pending a merge from azure-sdk-for-net, will update nuget packages properly when this occurs.
Please let me know if there are any other missing items from the PR while we wait.

**Comment by henry416** [comment]
@cormacpayne Are there any steps / files I am missing in this PR? (other than the help files)

**Comment by cormacpayne** [comment]
@azuresdkci add to whitelist

**Comment by henry416** [comment]
@cormacpayne can we rerun the check? It seems to be stuck.

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by henry416** [comment]
@cormacpayne I think the checks are failing b/c we haven't built and published a nuget package for the azure-sdk network branch. In the meantime is there any other component of this PR that is missing?

File: src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj [hunk 1]
```diff
@@ -80,8 +80,8 @@
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Compute.14.0.0-prerelease\lib\net45\Microsoft.Azure.Management.Compute.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=9.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.9.1.0-preview\lib\net45\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.9.2.0-preview\lib\net45\Microsoft.Azure.Management.Network.dll</HintPath>
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj]: @henry416 this will need to be bumped up to 10.0.0-preview

File: src/ResourceManager/Compute/Commands.Compute/packages.config [hunk 2]
```diff
@@ -9,7 +9,7 @@
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Authorization" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Compute" version="14.0.0-prerelease" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="9.1.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Network" version="9.2.0-preview" targetFramework="net45" />
```
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/packages.config]: @henry416 same comment

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 3]
```diff
@@ -63,8 +63,8 @@
     <Reference Include="Microsoft.Azure.Management.Authorization">
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Authorization.2.0.0\lib\net40\Microsoft.Azure.Management.Authorization.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=9.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.9.1.0-preview\lib\net45\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.9.2.0-preview\lib\net45\Microsoft.Azure.Management.Network.dll</HintPath>
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: @henry416 same comment

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj [hunk 4]
```diff
@@ -198,6 +198,7 @@
     <None Include="..\..\Common\Commands.ScenarioTests.ResourceManager.Common\AzureRM.Resources.ps1">
       <Link>ScenarioTests\AzureRM.Resources.ps1</Link>
     </None>
+    <None Include="app.config" />
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: @henry416 please remove this

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs [hunk 5]
```diff
@@ -47,7 +47,14 @@ public void TestVirtualNetworkGatewayConnectionWithBgpCRUD()
             NetworkResourcesController.NewInstance.RunPsTest("Test-VirtualNetworkGatewayConnectionWithBgpCRUD");
         }
 
-        [Fact]
+        [Fact(Skip = "Rerecord tests")]
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs]: @henry416 are these tests going to be re-recorded before this PR gets merged?

File: src/ResourceManager/Network/Commands.Network.Test/packages.config [hunk 6]
```diff
@@ -8,7 +8,7 @@
   <package id="Microsoft.Azure.Insights" version="0.13.1-preview" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Authorization" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Compute" version="14.0.0-prerelease" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="9.1.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Network" version="9.2.0-preview" targetFramework="net45" />
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network.Test/packages.config]: @henry416 same comment

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 7]
```diff
@@ -73,8 +73,8 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Authorization.2.0.0\lib\net40\Microsoft.Azure.Management.Authorization.dll</HintPath>
     </Reference>
-    <Reference Include="Microsoft.Azure.Management.Network, Version=9.1.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.9.1.0-preview\lib\net45\Microsoft.Azure.Management.Network.dll</HintPath>
+    <Reference Include="Microsoft.Azure.Management.Network, Version=9.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.9.2.0-preview\lib\net45\Microsoft.Azure.Management.Network.dll</HintPath>
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: @henry416 same comment

File: src/ResourceManager/Network/Commands.Network/Models/PSIpsecPolicy.cs [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Network.Models
+{
+    public class PSIpsecPolicy
+    {
+        public int SALifeTimeSeconds { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/Models/PSIpsecPolicy.cs]: @henry416 can you add a quick summary for each the properties?

```cs
/// <summary>
/// Brief description of property.
/// <summary>
public int SALifeTimeSeconds { get; set; }
```

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+            ipsecPolicy.SALifeTimeSeconds = this.SALifeTimeSeconds;
+            ipsecPolicy.SADataSizeKilobytes = this.SADataSizeKilobytes;
+            ipsecPolicy.IpsecEncryption = this.IpsecEncryption;
+            ipsecPolicy.IpsecIntegrity = this.IpsecIntegrity;
+            ipsecPolicy.IkeEncryption = this.IkeEncryption;
+            ipsecPolicy.IkeIntegrity = this.IkeIntegrity;
+            ipsecPolicy.DhGroup = this.DhGroup;
+            ipsecPolicy.PfsGroup = this.PfsGroup;
+
+            WriteObject(ipsecPolicy);
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs]: @henry416 as an FYI, StaticAnalysis is going to throw an exception because this cmdlet uses the `New` verb, but doesn't implement `ShouldProcess` . Since this cmdlet is creating a local object, you _do not_ need to implement this attribute, so you can suppress the exception. I will show you how to do so when the time comes 😄

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs [hunk 10]
```diff
@@ -130,6 +131,18 @@ public class NewAzureVirtualNetworkGatewayConnectionCommand : VirtualNetworkGate
             HelpMessage = "Do not ask for confirmation if you want to overrite a resource")]
         public SwitchParameter Force { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Whether to enable policy-based traffic selectors for a S2S connection")]
+        public string UsePolicyBasedTrafficSelectors { get; set; }
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs]: @henry416 why use a `string` here if you're going to just parse it into a `bool`? The recommended pattern is to add a parameter called `EnablePolicyBasedTrafficSelectors`, and when this switch is enabled, users can provide a value for the `IpsecPolicies` parameter. These two parameters would need to be in a separate parameter set from the default one, and `EnablePolicyBasedTrafficSelectors` would need to be a mandatory parameter. Let me know if you have any questions about this.

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+## SYNTAX
+
+```
+New-AzureRmIpsecPolicy -SALifeTimeSeconds <Int32> -SADataSizeKilobytes <Int32> -IpsecEncryption <String>
+ -IpsecIntegrity <String> -IkeEncryption <String> -IkeIntegrity <String> -DhGroup <String> -PfsGroup <String>
+```
+
+## DESCRIPTION
+{{Fill in the Description}}
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md]: @henry416 there are a few templates left behind from `platyPS` when generating the markdown for the first time. Would you mind replacing these templates (description, examples, etc.) with useful information for the user?

File: src/ResourceManager/Network/Commands.Network/packages.config [hunk 12]
```diff
@@ -9,7 +9,7 @@
   <package id="Microsoft.Azure.KeyVault.Core" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Authorization" version="1.0.0" targetFramework="net45" />
   <package id="Microsoft.Azure.Management.Compute" version="14.0.0-prerelease" targetFramework="net45" />
-  <package id="Microsoft.Azure.Management.Network" version="9.1.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Network" version="9.2.0-preview" targetFramework="net45" />
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/packages.config]: @henry416 same comment

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs (hunk 5 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs]: Is that the standard procedure? I can change both tests back to [Fact]

File: src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj (hunk 1 above)
Comment (henry416) [inline src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj]: @cormacpayne Shouldn't the version be 9.2.0? Anyways, the azure-sdk branch hasn't been merged yet so the nuget package to be published isn't available yet - think of the references as a placeholder for now.

File: src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj (hunk 4 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network.Test/Commands.Network.Test.csproj]: Will do.

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs (hunk 10 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs]: @cormacpayne This field is meant to act similar to the enableBgp field - it should be completely optional so that it doesn't break any existing PS scripts. PolicyBasedTrafficSelectors is also independent of the IpsecPolicies field.

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs (hunk 9 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs]: @cormacpayne Ah ok, that was an error that was confusing me. What would be the procedure to suppress that?

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md (hunk 11 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md]: @cormacpayne during platyPS regen, there were also a lot of other .md files changed that are not related to my work + Commands.Network/Microsoft.Azure.Commands.Network.dll-Help.xml - should I add those files as well?

File: src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj (hunk 1 above)
Comment (cormacpayne) [inline src/ResourceManager/Compute/Commands.Compute/Commands.Compute.csproj]: @henry416 it will be 10.0.0-preview after the changes that @jobatzil made

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs (hunk 9 above)
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs]: @henry416 we have a suppression `.csv` file for each of the StaticAnalysis analyzers that we have, so the one you will be concerned about is [`tools/StaticAnalysis/Exceptions/SignatureIssues.csv`](https://github.com/Azure/azure-powershell/blob/dev/tools/StaticAnalysis/Exceptions/SignatureIssues.csv), which is where you will paste the exception you want to suppress.

If you run the build locally, you will find a `SignatureIssues.csv` file in `src/Package/Debug` that contains all of the errors picked up during the build, and you will just need to copy the severity 1 issues from this file into the suppressed exceptions file mentioned previously.

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs (hunk 10 above)
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs]: @henry416 oops, sorry, I thought the two new parameters should be grouped together. In that case, you should just make this `string` parameter into a `SwitchParameter` called `EnablePolicyBasedTrafficSelectors`.

Following this pattern will make for a much better user experience - if the user wants to enable policy-based traffic selectors, then they will add `-EnablePolicyBasedTrafficSelectors` to the end of the command, and if they don't want to use it, then they don't have to do anything extra. Having the user provide a `string` that is being parsed into a `bool` is a less user-friendly way of having using a `SwitchParameter`.

Comment (henry416) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureVirtualNetworkGatewayConnectionCommand.cs]: @cormacpayne I have changed it to SwitchParameter

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs (hunk 9 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs]: @cormacpayne I have suppressed the exceptions for the files I have altered.

**Comment by henry416** [comment]
@cormacpayne Will I be able to push in changes after today? I haven't seen the sdk-for-net package being built yet - should I just put in the 10.0.0 version in the nuget references despite the package not existing?

**Comment by henry416** [comment]
@cormacpayne I have added auto-generated .md files - picked up several changes from other services that weren't in the checked-in files

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md (hunk 11 above)
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md]: @henry416 you do not need to add those files - I am working on a change that will get rid of the XML help from the repository and refresh all of the markdown files (updating them with the latest changes in the code), so all of that should be taken care of when my PR gets merged

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs (hunk 5 above)
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs]: @henry416 is there any reason you are skipping these tests?

File: src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md [hunk 13]
```diff
@@ -335,12 +344,15 @@ Creates a WAF configuration for an application gateway.
...
-### [New-AzureRmExpressRouteCircuit](New-AzureRmExpressRouteCircuit.md)
-Creates an Azure express route circuit.
+### [New-AzureRmExpressRouteCircuitAuthorization](New-AzureRmExpressRouteCircuitAuthorization.md)
+Creates an ExpressRoute circuit authorization.
 
 ### [New-AzureRmExpressRouteCircuitPeeringConfig](New-AzureRmExpressRouteCircuitPeeringConfig.md)
 Creates a new peering configuration to be added to an ExpressRoute circuit.
 
+### [New-AzureRmIpsecPolicy](New-AzureRmIpsecPolicy.md)
+{{Fill in the Synopsis}}
```
Comment (cormacpayne) [inline src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md]: @henry416 can you fill in this synopsis?

File: src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs (hunk 5 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network.Test/ScenarioTests/VirtualNetworkGatewayConnectionTests.cs]: I changed them back to [Fact]

**Comment by cormacpayne** [comment]
@henry416 we will make sure the SDK gets published by early next week, and then once it is published (and any other feedback for this PR is resolved), then we can merge this

**Comment by henry416** [comment]
@cormacpayne Okay, just to make sure, will this PR make the early april release in that case?

File: src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md (hunk 13 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network/help/AzureRM.Network.md]: @cormacpayne Done.

File: src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md (hunk 11 above)
Comment (henry416) [inline src/ResourceManager/Network/Commands.Network/help/New-AzureRmIpsecPolicy.md]: @cormacpayne  Synopsis is added.

**Comment by henry416** [comment]
@azuresdkci test this please

**Comment by henry416** [comment]
@cormacpayne I updated the nuget package. Pending sign off if nothing else is blocking.

**Comment by henry416** [comment]
@cormacpayne Fixed nuget package for compute

**Comment by henry416** [comment]
@cormacpayne I have fixed two tests, however, I think the application gateway test is failing - could you contact someone on app gw to fix this?

**Comment by jobatzil** [comment]
@henry416 Hey Henry :)
yes, we had breaking changes in the new azure-sdk which is the reason for the major version incrementation. I updated the Application Gateway related part in this PR: https://github.com/Azure/azure-powershell/pull/3678

**Comment by henry416** [comment]
@cormacpayne Looks clear then - can merge proceed?

**Comment by henry416** [comment]
@cormacpayne Fixed merge issues and added changelog

**Comment by henry416** [comment]
@cormacpayne Can this be merged?
